src/main/java/com/myhexin/cragent/coderabbittest/UserManager.java

```diff
@@ -0,0 +1,23 @@
+package com.myhexin.cragent.coderabbittest;
+
+import com.alibaba.fastjson.JSON;
+
+import java.util.Optional;
+
+public class UserManager {
+
+    /**
+     * 从客户信息 json 中获取身份证号
+     * {
+     *     "userInfo": {
+     *         "idCard": "xxx"
+     *     }
+     * }
+     *
+     * @param json 可能为 {@code null}
+     * @return 身份证号
+     */
+    private String getIdCard(String json) {
+        return Optional.ofNullable(json).map(JSON::parseObject).map(o -> o.getJSONObject("userInfo")).map(o -> o.getString("idCard")).orElse(null);
```

Comment:
**Issue**: The method getIdCard uses multiple map operations chained together, which can be hard to read and may lead to NullPointerException if any of the intermediate operations returns null. <br> **Fix**: Refactor the method to separate the operations into individual statements with proper null checks to enhance readability and maintainability. <br> **Code Suggestion**: 
 ```
 -        return Optional.ofNullable(json).map(JSON::parseObject).map(o -> o.getJSONObject("userInfo")).map(o -> o.getString("idCard")).orElse(null);
 +        if (json == null) return null;
 +        JSONObject jsonObject = JSON.parseObject(json);
 +        if (jsonObject == null) return null;
 +        JSONObject userInfo = jsonObject.getJSONObject("userInfo");
 +        if (userInfo == null) return null;
 +        return userInfo.getString("idCard");
 ```